src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -1410,6 +1647,12 @@ DatabaseShardImp::removeFailedShard(std::shared_ptr<Shard> shard)
         if (shard->index() == acquireIndex_)
             acquireIndex_ = 0;
 
+        if (shard->index() == latestShardIndex_)
+            latestShardIndex_ = boost::none;
+
+        if (shard->index() == secondLatestShardIndex_)
+            secondLatestShardIndex_ = boost::none;
+
```

Comment:
Not in this changeset, but it looks like the `shard` parameter should be passed by `const&` and not value (unless I'm missing the reason).

Reply:
It should be a reference.

Reply:
Done